app/src/main/java/com/weatherxm/ui/deviceforecast/ForecastDetailsViewModel.kt

```diff
@@ -80,13 +80,24 @@ class ForecastDetailsViewModel(
     }
 
     fun getSelectedDayPosition(selectedISODate: String?): Int {
-        return if (selectedISODate != null) {
-            val selectedLocalDate = LocalDate.parse(selectedISODate)
-            forecast.forecastDays.indexOfFirst {
-                it.date == selectedLocalDate
-            }
-        } else {
+        if (selectedISODate == null) {
+            return 0
+        }
+
+        val selectedLocalDate = LocalDate.parse(selectedISODate)
+        val position = forecast.forecastDays.indexOfFirst {
+            it.date == selectedLocalDate
+        }
+
+        return if (position == -1) {
+            /**
+             * Temporary code so we can figure out why some crashes occur, in which cases/dates
+             */
+            val allDates = forecast.forecastDays.joinToString(" : ") { it.date.toString() }
+            Timber.e("Could not find selected day in forecast: $selectedISODate $allDates")
```

Comment:
Maybe we should include the `selectedLocalDate` as well

Reply:
Valid point, done!